pkg/report/predicate/vuln.go

```diff
@@ -0,0 +1,91 @@
+package predicate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/package-url/packageurl-go"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/clock"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+// CosignVulnPredicate represents the Cosign Vulnerability predicate.
+// Cosign provides the CosignVulnPredicate structure in their repository.
+// But the type of Scanner.Result is defined as map[string]interface{}, which is difficult to use,
+// so we define our own.
+// The PR is in progress to replace Scanner.Result type to interface{}.
+// https://github.com/sigstore/cosign/pull/2096
+type CosignVulnPredicate struct {
+	Invocation Invocation `json:"invocation"`
+	Scanner    Scanner    `json:"scanner"`
+	Metadata   Metadata   `json:"metadata"`
+}
+
+type Invocation struct {
+	Parameters interface{} `json:"parameters"`
+	URI        string      `json:"uri"`
+	EventID    string      `json:"event_id"`
+	BuilderID  string      `json:"builder.id"`
+}
+
+type DB struct {
+	URI     string `json:"uri"`
+	Version string `json:"version"`
+}
+
+type Scanner struct {
+	URI     string       `json:"uri"`
+	Version string       `json:"version"`
+	DB      DB           `json:"db"`
+	Result  types.Report `json:"result"`
+}
+
+type Metadata struct {
+	ScanStartedOn  time.Time `json:"scanStartedOn"`
+	ScanFinishedOn time.Time `json:"scanFinishedOn"`
+}
+
+type Writer struct {
```

Comment:
We may have more predicates in the future.

```suggestion
type VulnWriter struct {
```

Reply:
Done